Source/Dafny/Compiler-Csharp.cs

```diff
@@ -705,17 +720,28 @@ protected class ClassWriter : IClassWriter {
         }
       }
 
+      var customReceiver = !m.IsStatic && m.EnclosingClass is NewtypeDecl;
```

Comment:
This repeats 5 times in the review.  Could we add it to `Method` perhaps?

Reply:
Good idea.